config/src/main/java/org/springframework/security/config/annotation/web/configurers/ott/OneTimeTokenLoginConfigurer.java

```diff
@@ -0,0 +1,369 @@
+		this.oneTimeTokenGenerateRequestResolver = oneTimeTokenGenerateRequestResolver;
+		return this;
+	}
+
+	/**
+	 * Specifies the {@link AuthenticationProvider} to use when authenticating the user.
+	 * @param authenticationProvider
+	 */
+	public OneTimeTokenLoginConfigurer<H> authenticationProvider(AuthenticationProvider authenticationProvider) {
+		Assert.notNull(authenticationProvider, "authenticationProvider cannot be null");
+		this.authenticationProvider = authenticationProvider;
+		return this;
+	}
+
+	/**
+	 * Specifies the URL that a One-Time Token generate request will be processed.
+	 * Defaults to {@code POST /ott/generate}.
+	 * @param generateUrl
+	 */
+	public OneTimeTokenLoginConfigurer<H> generateUrl(String generateUrl) {
+		Assert.hasText(generateUrl, "generateUrl cannot be null or empty");
+		this.generateUrl = generateUrl;
+		return this;
+	}
+
+	/**
+	 * Specifies strategy to be used for successful generated one-time tokens. By default,
+	 * a redirect will be performed to {@code POST /login/ott} using the
+	 * {@link RedirectGeneratedOneTimeTokenSuccessHandler}. It is often needed to provide
+	 * your own implementation of this interface so the one-time token is also delivered
+	 * to the user.
+	 * @param generatedOneTimeTokenSuccessHandler
+	 */
+	public OneTimeTokenLoginConfigurer<H> generatedOneTimeTokenSuccessHandler(
+			GeneratedOneTimeTokenSuccessHandler generatedOneTimeTokenSuccessHandler) {
+		Assert.notNull(generatedOneTimeTokenSuccessHandler, "generatedOneTimeTokenSuccessHandler cannot be null");
+		this.generatedOneTimeTokenSuccessHandler = generatedOneTimeTokenSuccessHandler;
+		return this;
+	}
+
+	/**
+	 * Specifies the URL to process the login request, defaults to {@code /login/ott}.
+	 * Only POST requests are processed, for that reason make sure that you pass a valid
+	 * CSRF token if CSRF protection is enabled.
+	 * @param loginProcessingUrl
+	 * @see org.springframework.security.config.annotation.web.builders.HttpSecurity#csrf(Customizer)
+	 */
+	public OneTimeTokenLoginConfigurer<H> loginProcessingUrl(String loginProcessingUrl) {
+		Assert.hasText(loginProcessingUrl, "loginProcessingUrl cannot be null or empty");
+		this.loginProcessingUrl = loginProcessingUrl;
+		return this;
+	}
+
+	/**
+	 * Configures whether the default one-time token submit page should be shown. This
+	 * will prevent the {@link DefaultOneTimeTokenSubmitPageGeneratingFilter} to be
+	 * configured.
+	 * @param show
+	 */
+	public OneTimeTokenLoginConfigurer<H> showSubmitPage(boolean show) {
+		this.submitPageEnabled = show;
+		return this;
+	}
+
+	/**
+	 * Sets the URL that the default submit page will be generated. Defaults to
+	 * {@code /login/ott}. Note that if you don't want to generate the default submit page
+	 * you should use {@link #showSubmitPage(boolean)}.
+	 * @param submitPageUrl
+	 */
+	public OneTimeTokenLoginConfigurer<H> submitPageUrl(String submitPageUrl) {
```

Comment:
I went with `defaultSubmitPageUrl` because I think there will be some confusion when using the word `generate` about whether it's the generated page or generated token.